### PR TITLE
Fix no-engine

### PR DESCRIPTION
**Pull request by mattcaswell**
If we specify no-engine then this should cascade to also mean
no-dynamic-engine. The store test was only checking whether
dynamic-engine was disabled, meaning that some tests were failing
in a no-engine build.

### CODE REVIEW

**Review by t8m — APPROVED** [review]

**Comment by mattcaswell** [comment]
Err...not sure why this has been labelled as urgent? Removing that label.

**Comment by kaduk** [comment]
> Err...not sure why this has been labelled as urgent? Removing that label.

It fixes the tests in a no-engine build?  I thought that counted as a build break, thus urgent

**Review by kaduk — COMMENTED** [review]
What about `static-engine`?

**Comment by mattcaswell** [comment]
> It fixes the tests in a no-engine build? I thought that counted as a build break, thus urgent

But not in a standard (default) build, nor any run by our standard CIs. The only thing that notices is run-checker - and that's been broken for months and no one noticed. So I don't think this can qualify as urgent. It's not stopping anyone from raising PRs or doing their work. I'm fairly sure I had my wrists slapped a while ago for pushing through a run-checker fix as urgent.

File: Configure [hunk 1]
```diff
@@ -581,7 +581,7 @@ my @disable_cascades = (
 
     "module"            => [ "fips", "dso" ],
 
-    "engine"            => [ grep /eng$/, @disablables ],
+    "engine"            => [ grep /(eng$|dynamic-engine)/, @disablables ],
```
Comment (levitte) [inline Configure:584]: Uhm, that looks overly incomprehensible.  Why not this?

``` perl
    "engine"            => [ "dynamic-engine", grep(/eng$/, @disablables) ],
```

Comment (mattcaswell) [reply Configure]: Yeah. That works too. I've changed it to look like that.

**Comment by mattcaswell** [comment]
Please reconfirm?

**Review by levitte — APPROVED** [review]

**Comment by t8m** [comment]
Still approved. I added urgent as I thought this affects the minimal build.

**Comment by kaduk** [comment]
I don't see an answer to my question about static-engine

**Comment by mattcaswell** [comment]
> I don't see an answer to my question about static-engine

Sorry, missed that. I pushed a fixed up adding "static-engine" to the cascades. Although, in practice its not a problem since everything that checks that is already also guarded by a "no-engine" check.

**Review by kaduk — APPROVED** [review]
The line is kind of long now, but maybe that's okay.

File: Configure [hunk 2]
```diff
@@ -581,7 +581,7 @@ my @disable_cascades = (
 
     "module"            => [ "fips", "dso" ],
 
-    "engine"            => [ grep /eng$/, @disablables ],
+    "engine"            => [ "dynamic-engine", "static-engine", grep(/eng$/, @disablables) ],
```
Comment (levitte) [inline Configure:584]: No!

`static-engine` is handled further down in Configure, as a special case.  `%disabled` shall *only* have `dynamic-engine`, or not.  The rest is handled properly here:

https://github.com/openssl/openssl/blob/2d840893e78253bcce428603fdbcda159bdebe08/Configure#L1454-L1458

and here:

https://github.com/openssl/openssl/blob/2d840893e78253bcce428603fdbcda159bdebe08/Configure#L1784-L1788

**Review by levitte — CHANGES_REQUESTED** [review]

**Comment by levitte** [comment]
Sorry that I didn't catch the `static-engine` question earlier as well...

**Comment by mattcaswell** [comment]
Ah! Ok. Thanks for the clarification. I have dropped the last commit.

@levitte, since this is now back in the state where you previously approved it, I'm assuming your approval still stands.

**Comment by levitte** [comment]
> @levitte, since this is now back in the state where you previously approved it, I'm assuming your approval still stands.

Yes

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**